application/views/helpers/FileMarkup.php

```diff
@@ -797,19 +797,14 @@ public function image_tag($record, $props, $format)
         if (!$record) {
             return false;
         }
-        
-        if ($record instanceof File) {
-            $filename = $record->getDerivativeFilename();
-            $file = $record;
-        } else if ($record instanceof Item) {
-            $item = $record;
-            $file = get_db()->getTable('File')->getRandomFileWithImage($item->id);
-            if (!$file) {
-                return false;
-            }
-            $filename = $file->getDerivativeFilename();
-        } else {
-            // throw some exception?
+
+        // Use the representative file.
+        if (!method_exists($record, 'getFile')) {
```

Comment:
`getFile` is defined for all records, so I think it's better to do an argument typehint on `Omeka_Record_AbstractRecord` or a `instanceof` check for that class rather than duck-typing on the method name here.